File: src/domain/js/entities.js
/**
 * Domain Entities — JS Edition
 */

export class ProgressEvent {
  constructor({ userId, delta, timestamp }) {
    this.userId = userId;
    this.delta = delta;
    this.timestamp = timestamp;
  }
}

export class HealthScore {
  constructor({ value }) {
    this.value = value;
  }
}

export class RehabMilestone {
  constructor({ step, completed }) {
    this.step = step;
    this.completed = completed;
  }
}

export class UserProfile {
  constructor({ userId, region }) {
    this.userId = userId;
    this.region = region;
  }
}

export class ZKProofEnvelope {
  constructor({ proofHash, metadata }) {
    this.proofHash = proofHash;
    this.metadata = metadata;
  }
}

File: src/domain/python/entities.py
"""
Domain Entities — Python Edition
"""

class ProgressEvent:
    def __init__(self, user_id, delta, timestamp):
        self.user_id = user_id
        self.delta = delta
        self.timestamp = timestamp

class HealthScore:
    def __init__(self, value: float):
        self.value = value

class RehabMilestone:
    def __init__(self, step: int, completed: bool):
        self.step = step
        self.completed = completed

class UserProfile:
    def __init__(self, user_id: str, region: str):
        self.user_id = user_id
        self.region = region

class ZKProofEnvelope:
    def __init__(self, proof_hash: str, metadata: dict):
        self.proof_hash = proof_hash
        self.metadata = metadata
